source/utilities.tex

```diff
@@ -15967,6 +15972,399 @@
+that generalize the notion of a callable object\iref{func.def}.
+These wrappers can store, move, and call arbitrary callable objects,
+given a call signature, allowing functions to be first-class objects.
+
+\pnum
+Implementations are encouraged to avoid the use of dynamically allocated memory
+for a small contained value.
+\begin{note}
+Such small-object optimization can only be applied to types \tcode{T}
+for which \tcode{is_nothrow_move_constructible_v<T>} is \tcode{true}.
+\end{note}
+
+\rSec4[func.wrap.move.ctor]{Constructors and destructor}
+
+\indexlibraryctor{move_only_function}%
+\begin{itemdecl}
+template<class VT>
+  static constexpr bool @\exposid{is-callable-from}@ = @\seebelow@;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+If \placeholder{noex} is \tcode{true},
+\tcode{\exposid{is-callable-from}<VT>} is equal to
+\tcode{is_nothrow_invocable_r_v<R, VT \cv{} \placeholder{ref}, Args...> \&\&
+is_nothrow_invocable_r_v<R, VT \placeholder{inv-quals}, Args...>}.
+Otherwise, \tcode{\exposid{is-callable-\linebreak from}<VT>} is equal to
+\tcode{is_invocable_r_v<R, VT \cv{} \placeholder{ref}, Args...> \&\&
+is_invocable_r_v<R, VT \placeholder{inv-quals}, Args...>}.
+\end{itemdescr}
+
+\indexlibraryctor{move_only_function}%
+\begin{itemdecl}
+move_only_function() noexcept;
+move_only_function(nullptr_t) noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\ensures
+\tcode{*this} has no target object.
+\end{itemdescr}
+
+\indexlibraryctor{move_only_function}%
+\begin{itemdecl}
+move_only_function(move_only_function&& f) noexcept;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\ensures
+The target object of \tcode{*this} is
+the target object \tcode{f} had before construction, and
+\tcode{f} is in a valid state with an unspecified value.
+\end{itemdescr}
+
+\indexlibraryctor{move_only_function}%
+\begin{itemdecl}
+template<class F> move_only_function(F&& f);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+Let \tcode{VT} be \tcode{decay_t<F>}.
+
+\pnum
+\constraints
+\begin{itemize}
+\item
+\tcode{remove_cvref_t<F>} is not the same type as \tcode{move_only_function}, and
+\item
+\tcode{remove_cvref_t<F>} is not a specialization of \tcode{in_place_type_t}, and
+\item
+\tcode{\exposid{is-callable-from}<VT>} is \tcode{true}.
+\end{itemize}
+
+\pnum
+\mandates
+\tcode{is_constructible_v<VT, F>} is \tcode{true}.
+
+\pnum
+\expects
+\tcode{VT} meets the \oldconcept{Destructible} requirements, and
+if \tcode{is_move_constructible_v<VT>} is \tcode{true},
+\tcode{VT} meets the \oldconcept{MoveConstructible} requirements.
+
+\pnum
+\ensures
+\tcode{*this} has a target object unless any of the following hold:
+\begin{itemize}
+\item
+\tcode{f} is a null function pointer value, or
+\item
+\tcode{f} is a null member pointer value, or
+\item
+\tcode{remove_cvref_t<F>} is a specialization of
+the \tcode{move_only_function} class template,
+and \tcode{f} has no target object.
+\end{itemize}
+Otherwise, \tcode{*this} targets an object of type \tcode{VT}
```

Comment:
I feel like I read a comment calling out that this uses "target" wrong. I think it also listed some other defects. I can't find it, though.

Reply:
There are ~2 similarly wrong uses of "target" below.

Reply:
Ah, there it is: https://github.com/cplusplus/draft/issues/4960.

Reply:
Yes, I thought I'd already linked to it here, but I did it at https://github.com/cplusplus/draft/issues/4977#issuecomment-939113451 instead 🤦‍♂️